kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java

```diff
@@ -1312,7 +1333,11 @@ XmlComponentConfigurations loadEncryptedSnapshotFileContent(long snapshotID) thr
         }
 
         // File loaded, try to decrypt and unmarshall
-        String decryptedContent = new String(this.m_cryptoService.decryptAes(entireFile.toString().toCharArray()));
+        char[] decryptAes = this.m_cryptoService.decryptAes(entireFile.toString().toCharArray());
+        if (decryptAes == null) {
```

Comment:
So, if decryptAes fails to decrypt the char array it returns null?
Ideally it should throw an exception. If not (and we don't want to break the API) this should be documented in the Javadoc.